mbuild/tests/test_cassandramcf.py

```diff
@@ -0,0 +1,273 @@
+        oplsaa = foyer.Forcefield(name='oplsaa')
+        ethane = oplsaa.apply(ethane)
+        with pytest.raises(ValueError,match=r'Unreasonable value'):
+            mb.formats.cassandramcf.write_mcf(ethane,'ethane.mcf',
+                    angle_style='harmonic',dihedral_style='opls',lj14=2.0)
+
+    @pytest.mark.skipif(not has_foyer, reason="Foyer package not installed")
+    def test_unreasonable_coul14(self,ethane):
+        import foyer
+        oplsaa = foyer.Forcefield(name='oplsaa')
+        ethane = oplsaa.apply(ethane)
+        with pytest.raises(ValueError,match=r'Unreasonable value'):
+            mb.formats.cassandramcf.write_mcf(ethane,'ethane.mcf',
+                    angle_style='harmonic',dihedral_style='opls',coul14=-1.0)
+
+    @pytest.mark.skipif(not has_foyer, reason="Foyer package not installed")
+    def test_multiple_molecules(self, ethane):
+        n_ethane = 2
+        ethane.name = 'Ethane'
+        filled = mb.fill_box(ethane,n_compounds=n_ethane,box=[0,0,0,4,4,4])
+        with pytest.raises(ValueError,match=r'Not all components of the molecule'):
+            filled.save(filename='box-ethane-opls.mcf', forcefield_name='oplsaa', 
+                    angle_style='harmonic', dihedral_style='opls')
+
+    @pytest.mark.skipif(not has_foyer, reason="Foyer package not installed")
+    def test_save_forcefield(self, ethane):
+        ethane.save(filename='ethane-opls.mcf', forcefield_name='oplsaa', 
+                angle_style='harmonic', dihedral_style='opls')
+
+
+        mcf_data = []
+        with open('ethane-opls.mcf') as f:
+            for line in f:
+                mcf_data.append(line.strip().split())
+
+        for idx,line in enumerate(mcf_data):
+            if len(line) > 1:
+                if line[1] == 'Atom_Info':
+                    atom_section_start = idx
+                elif line[1] == 'Bond_Info':
+                    bond_section_start = idx
+                elif line[1] == 'Angle_Info':
+                    angle_section_start = idx
+                elif line[1] == 'Dihedral_Info':
+                    dihedral_section_start = idx
+                elif line[1] == 'Improper_Info':
+                    improper_section_start = idx
+                elif line[1] == 'Fragment_Info':
+                    fragment_section_start = idx
+                elif line[1] == 'Fragment_Connectivity':
+                    fragment_conn_start = idx
+
+        # Check a some atom info
+        assert mcf_data[atom_section_start+1][0] == '8'
+        assert mcf_data[atom_section_start+2][1] == 'ls_135'
+        assert mcf_data[atom_section_start+2][3] == '12.011'
+        assert mcf_data[atom_section_start+2][4] == '-0.180'
+        assert mcf_data[atom_section_start+2][5] == 'LJ'
+        assert mcf_data[atom_section_start+2][6] == '33.212'
+        assert mcf_data[atom_section_start+2][7] == '3.500'
+
+        # Bond info
+        assert mcf_data[bond_section_start+1][0] == '7'
+        passed_test = False
```

Comment:
I think this bool/break logic could be removed? The asserts should handle the cases of it failing or not and unless the files are really long, iterating through useless lines is pretty quick

Reply:
The logic is there because I need to make sure the atom ids participating in the bond are correct before I check the bond params. The assert statements are only executed once the atom id condition is met. In principle, the atom id condition could never be met and then the assert statements would never be executed.